DocCRender.docc/contributing/Internals.md

```diff
@@ -0,0 +1,189 @@
+        slot-scope="{ url, title }"
+        :to="url"
+    >
+        {{ title }}
+    </router-link>
+</ReferenceUrlProvider>
+```
+
+### Stores
+
+A DocC Render store module extracts commonly used data, that needs to be reactive and be accessible to multiple components. A store is just an ES Module, that exposes a `state` object and some methods.
+
+```js
+// src/stores/FooStore.js
+export default {
+  state: {},
+  setFoo(foo){
+    this.state.foo = foo;  
+  },
+}
+```
+
+In order to make the state reactive, a Vue component imports the module and attaches it's `state` to a `data` property.
+
+```js
+// src/views/FooView.vue
+
+import FooStore from '@/stores/FooStore'
+
+export default {
+  computed: {
+    store: FooStore
+  },
+  provide() {
+    return { store: this.store };
+  },
+}
+```
+
+```js
+// src/components/FooView/FooComponent.vue
+ 
+export default {
+  inject: ['store'],
+  data() {
+    return {
+      state: this.store.state,
+    }
+  }
+}
+```
+
+### Constants
+
+Files inside the `constants` folder are meant to share different constant value across many components. They are preferably kept simple, scoping common constants to a single file.
+
+### Icons
+
+Icons in DocC Render are Vue files, that implement the `SVGIcon` component. Inside the `SVGIcon` element, you copy the body of an SVG file, with the `svg` tag removed. This is done, so we can load vector icons inline, and be able to change properties like colors and fills easily with css.
+
+```html
+<template>
+  <SVGIcon class="diagonal-arrow" viewBox="0 0 14 14">
+    <path d="M0.010 12.881l10.429-10.477-3.764 0.824-0.339-1.549 7.653-1.679-1.717 7.622-1.546-0.349 0.847-3.759-10.442 10.487z"></path>
+  </SVGIcon>
+</template>
+
+<script>
+import SVGIcon from '@/components/SVGIcon.vue';
+
+export default {
+  name: 'DiagonalArrowIcon',
+  components: { SVGIcon },
+};
+</script>
+``` 
+
+### Styles
+
+Styles in DocC Render are separated between `core` and `base` styles, with the majority of styling being defined as scoped inside the corresponding Vue file.
+
+> Define styles inside Vue files as **scoped**, so style collisions are reduced to a minimum. 
+
+* The `base` styles represent global styles, like resetting styles, styling links and headings across the app. These styles are imported once, none scoped, inside App.vue.
+* The `core` styles are used to define Sass variables, mixins and functions, that are to be used to generate reusable and complex styles inside Vue files. 
+
+> Important: SCSS files inside **core** should never output css directly! These style files are imported in almost every vue file, and making them output styles, will lead to duplication of styles across the entire app. 
+
+If you need access to a global mixin or var, from `_core.scss` you have to import it first.
+
+```html
+<style scoped lang="scss">
+    @import '@/styles/_core.scss';
+    .some-class {
+        margin-left: $varfrom-core-folder;
+    }
+</style>
+```
+
+## Unit tests
```

Comment:
Can we use an h3 for this (and maybe h4 for its sub-items)? Or have you considered separating this article out into separate sub articles? (Not sure if it would work well or not, just wondering)

Reply:
I could write a more detailed guide, but I was not sure how detailed we want it to be.